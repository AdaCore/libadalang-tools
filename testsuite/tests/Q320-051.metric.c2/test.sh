cd src
gnatmetric -v --all -x -sfn *.ad? > stdout
cat metrix.xml stdout *.metrix
