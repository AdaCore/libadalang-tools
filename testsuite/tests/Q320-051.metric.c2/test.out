Line metrics summed over 16 units
  all lines            : 277
  code lines           : 128
  comment lines        : 105
  end-of-line comments : 0
  comment percentage   : 45.06
  blank lines          : 44

Average lines in body: 8.33

Element metrics summed over 16 units
  all statements      : 39
  all declarations    : 31
  logical SLOC        : 70

 0 public types in 0 units
 including
    0 private types

 unknown public subprograms in 14 units

 11 subprogram bodies in 11 units

 0 type declarations in 0 units

Average cyclomatic complexity: 1.75

Coupling metrics:
=================
   Unit C23001A (c23001a.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 0

   Unit C230063PROC (c230063proc.ads)
      unit fan-out coupling     : 0
      unit fan-in coupling      : 0

   Unit C23006A (c23006a.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 0

   Unit C23006B (c23006b.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 2
      unit fan-in coupling      : 0

   Unit C23006BPKG (c23006bpkg.ads)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006B_PKG (c23006b_pkg.ads)
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006C (c23006c.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 4
      unit fan-in coupling      : 0

   Unit C23006CFUNC (c23006cfunc.ads)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006CPROC (c23006cproc.ads)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006C_FUNC (c23006c_func.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006C_PROC (c23006c_proc.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 1

   Unit C23006D (c23006d.adb)
      control fan-out coupling  : 0
      control fan-in coupling   : 0
      unit fan-out coupling     : 0
      unit fan-in coupling      : 0
